src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -471,7 +479,9 @@ inline oop ShenandoahHeap::try_evacuate_object(oop p, Thread* thread, Shenandoah
   oop result = ShenandoahForwarding::try_update_forwardee(p, copy_val);
   if (result == copy_val) {
     // Successfully evacuated. Our copy is now the public one!
-    _evac_tracker->end_evacuation(thread, size * HeapWordSize, ShenandoahHeap::get_object_age(copy_val));
+    // If copying to the old generation, we don't care about recording
+    // object age in the census stats.
+    _evac_tracker->end_evacuation(thread, size * HeapWordSize, young_gen, young_gen ? ShenandoahHeap::get_object_age(copy_val) : 0);
```

Comment:
Revisit this to see why we record age 0 for the case of old gen objects.

Reply:
To answer the question, "why 0?" : it was because I didn't want to extract the age for the old_gen case, because it isn't used in that case (the census doesn't record old gen object ages). I'll see about cleaning this up a bit.

Reply:
Cleaned this up to avoid this pattern. Hopefully, it reads a bit cleaner and less clever (and error-prone) now.